app/bundles/EmailBundle/EventListener/TokenSubscriber.php

```diff
@@ -73,23 +74,38 @@ public function onTokenReplacement(TokenReplacementEvent $event)
         }
 
         $lead      = $event->getLead();
+        $tokens    = $clickthrough['tokens'];
```

Comment:
This will cause problems if the 'tokens' key doesn't exist. The `EmailEvents::TOKEN_REPLACEMENT` event is triggered in more places in Mautic. For example in the CampaignBundle.